src/app/GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1612,6 +1612,17 @@ private void Unstage(bool canUseUnstageAll = true)
                         unstagedFiles[index].IsDeleted = item.IsDeleted;
                         unstagedFiles[index].IsTracked = item.IsTracked;
                         unstagedFiles[index].IsChanged = item.IsChanged;
+                        GitItemStatus unstagedItem = unstagedFiles[index];
+                        if (unstagedItem.IsSubmodule)
+                        {
+                            // existing submodule, update its status, may be dirty
+                            unstagedItem.SetSubmoduleStatus(ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                            {
+                                return await SubmoduleHelpers.GetSubmoduleCurrentChangesAsync(Module, unstagedItem.Name, unstagedItem.OldName, unstagedItem.Staged == StagedStatus.Index)
+                                    .ConfigureAwait(false);
+                            }));
+                        }
+
```

Comment:
Rather make `GitModule.GetSubmoduleCurrentStatus` accessible than tripling this complex call (with ugly ConfigureAwait).

```suggestion
                        Module.GetSubmoduleCurrentStatus([unstagedFiles[index]]);
```